broker/data-access-layer/iaas/test/iaas.AwsClient.spec.js

```diff
@@ -0,0 +1,293 @@
+            expect(disk.size).to.eql('4');
+            expect(disk.zone).to.eql('zone');
+            expect(disk.type).to.eql('type');
+            expect(disk.extra).to.deep.equal({
+              type: 'type',
+              tags: {
+                k1: 'v1',
+                k2: 'v2'
+              }
+            });
+          });
+      });
+      it('should fail on creating disk from snapshot for aws', function () {
+        const client = new AwsClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const createVolStub = sinon.stub(client.blockstorage, 'createVolume');
+        createVolStub.withArgs({
+          AvailabilityZone: zone,
+          SnapshotId: snapshotId,
+          VolumeType: volType,
+          TagSpecifications: [{
+            ResourceType: 'volume',
+            Tags: [{
+              Key: 'customkey',
+              Value: 'customvalue'
+            }, {
+              Key: 'createdBy',
+              Value: 'service-fabrik'
+            }]
+          }]
+        }).returns({
+          promise: () => Promise.reject(new Error('diskerror'))
+        });
+        return client
+          .createDiskFromSnapshot(snapshotId, zone, {
+            volumeType: 'gp2',
+            tags: {
+              customkey: 'customvalue'
+            }
+          })
+          .catch(err => {
+            expect(err.message).to.eql('diskerror');
+          });
+      });
+      it('should fail on waiting for disk from snapshot for aws', function () {
+        const client = new AwsClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const createVolStub = sinon.stub(client.blockstorage, 'createVolume');
+        createVolStub.withArgs({
+          AvailabilityZone: zone,
+          SnapshotId: snapshotId,
+          VolumeType: volType,
+          TagSpecifications: [{
+            ResourceType: 'volume',
+            Tags: [{
+              Key: 'customkey',
+              Value: 'customvalue'
+            }, {
+              Key: 'createdBy',
+              Value: 'service-fabrik'
+            }]
+          }]
+        }).returns({
+          promise: () => Promise.resolve(response)
+        });
+        const waitForStub = sinon.stub(client.blockstorage, 'waitFor');
+        waitForStub.withArgs('volumeAvailable', {
+          VolumeIds: [diskId]
+        })
+          .returns({
+            promise: () => Promise.reject(new Error('diskwaiterror'))
+          });
+        return client
+          .createDiskFromSnapshot(snapshotId, zone, {
+            volumeType: 'gp2',
+            tags: {
+              customkey: 'customvalue'
+            }
+          })
+          .catch(err => {
+            expect(err.message).to.eql('diskwaiterror');
+          });
+      });
+    });
+
+    describe('#getDiskMetadata', function () {
```

Comment:
Took this entirely from `iaas.ClouProviderClient.js` except changed the `class` used to create the client.